ports/esp8266/mpconfigport.h

```diff
@@ -116,7 +116,11 @@
 #define MICROPY_FATFS_LFN_CODE_PAGE    437 /* 1=SFN/ANSI 437=LFN/U.S.(OEM) */
 #define MICROPY_ESP8266_APA102         (1)
 
-#define MICROPY_EVENT_POLL_HOOK {ets_event_poll();}
+// No blocking wait-for-event on ESP8266, only non-blocking poll of the OS event
+// loop
+#define MICROPY_INTERNAL_WFE(TIMEOUT_MS)
+#define MICROPY_INTERNAL_EVENT_HOOK ets_event_poll()
```

Comment:
The `ets_event_poll()` function is implemented as:
```c
void ets_event_poll(void) {
    ets_loop_iter();
    mp_handle_pending(true);
}
```
It already calls `mp_handle_pending(true)`, so now will call it twice in `mp_event_xxx()` functions.

I suggest changing this macro to call `ets_loop_iter()` directly, remove `ets_event_poll()`, and for the small number of uses of `ets_event_poll()` change them to `mp_event_handle_nowait()`.